plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergReadVersionedTable.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+
+import static io.trino.plugin.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static java.lang.String.format;
+
+public class TestIcebergReadVersionedTable
+        extends AbstractTestQueryFramework
+{
+    private long v1Unixtime;
+    private long v1SnapshotId;
+    private long v2Unixtime;
+    private long v2SnapshotId;
+    private long incorrectSnapshotId;
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createIcebergQueryRunner();
+    }
+
+    @BeforeClass
+    public void setUp() throws InterruptedException
+    {
+        assertQuerySucceeds("CREATE TABLE test_iceberg_read_versioned_table (_string VARCHAR, _integer INTEGER)");
+        assertQuerySucceeds("INSERT INTO test_iceberg_read_versioned_table VALUES ('a', 1)");
+        v1Unixtime = Instant.now().getEpochSecond() + 1; // go to the next second in case the operation finishes in the same second
+        v1SnapshotId = getLatestSnapshotId("test_iceberg_read_versioned_table");
```

Comment:
v1SnapshotId is the ID of the v1 snapshot
v1Unixtime should be timestamp of the v1 snapshot, not "some timestamp after that"
read actual snapshot's timestamp (`committed_at`) instead of assuming (otherwise code requires explanation)